app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/NrsService.scala

```diff
@@ -0,0 +1,139 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.excisemovementcontrolsystemapi.services
+
+import play.api.Logging
+import play.api.http.Status.INTERNAL_SERVER_ERROR
+import uk.gov.hmrc.auth.core.authorise.EmptyPredicate
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals
+import uk.gov.hmrc.auth.core.retrieve._
+import uk.gov.hmrc.auth.core.{AffinityGroup, AuthConnector, ConfidenceLevel, CredentialRole}
+import uk.gov.hmrc.excisemovementcontrolsystemapi.connectors.NrsConnector
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.auth.ValidatedXmlRequest
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.nrs.{IdentityData, NonRepudiationSubmissionAccepted, NrsMetadata, NrsPayload}
+import uk.gov.hmrc.excisemovementcontrolsystemapi.services.NrsService.nonRepudiationIdentityRetrievals
+import uk.gov.hmrc.excisemovementcontrolsystemapi.utils.{DateTimeService, EmcsUtils, ErnsMapper, NrsEventIdMapper}
+import uk.gov.hmrc.http.{Authorization, HeaderCarrier, InternalServerException}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+
+@Singleton
+class NrsService @Inject()
+(
+  authConnector: AuthConnector,
+  nrsConnector: NrsConnector,
+  dateTimeService: DateTimeService,
+  emcsUtils: EmcsUtils,
+  ernsMapper: ErnsMapper,
+  nrsEventIdMapper: NrsEventIdMapper
+)(implicit ec: ExecutionContext) extends Logging {
+
+  def submitNrs(
+    request: ValidatedXmlRequest[_],
+    correlationId: String
+  )(implicit headerCarrier: HeaderCarrier): Future[Either[Int, NonRepudiationSubmissionAccepted]] = {
+
+    val payload = request.body.toString
+    val userHeaderData = request.headersAsMap
+    val message = request.message
+    val exciseNumber = ernsMapper.getSingleErnFromMessage(message, request.validErns)
+    val notableEventId = nrsEventIdMapper.mapMessageToEventId(message)
+
+    (for {
+      identityData <- retrieveIdentityData()
+      userAuthToken = retrieveUserAuthToken(headerCarrier)
+      metaData = NrsMetadata.create(payload, emcsUtils,notableEventId, identityData, dateTimeService.nowUtc.toString,
+        userAuthToken, userHeaderData, exciseNumber)
+      encodedPayload = emcsUtils.encode(payload)
+      nrsPayload = NrsPayload(encodedPayload, metaData)
+      retrievedNrsResponse <- nrsConnector.sendToNrs(nrsPayload, correlationId)
+    } yield retrievedNrsResponse)
+      .recover {
+        case NonFatal(e) =>
+          logger.warn(s"[NrsService] - Error when submitting to Non repudiation system (NRS) with message: ${e.getMessage}", e)
+          Left(INTERNAL_SERVER_ERROR)
```

Comment:
The failure on this call i.e. the Left(Int) response is not used apart from the tests, should we just refactor this to just return a NonRepudiationSubmissionAccepted?

Reply:
done